inspec/a2-api-integration/controls/iam_v2.rb

```diff
@@ -427,63 +428,207 @@
+    TEST_USER = {
+        id: CUSTOM_USER_ID,
+        name: "display name !#$#",
+        password: "chefautomate"
+    }
+
+    describe "when the user is not yet created" do
+      CREATED_ID = "unique-user-id"
+
+      after(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users/#{CREATED_ID}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+      end
+
+      describe "POST /iam/v2beta/users/:id" do
+        it "creates the user" do
+          createdName =  "i created my own name"
+          resp = automate_api_request("/apis/iam/v2beta/users",
+            http_method: 'POST',
+            request_body: {
+              id: CREATED_ID,
+              name: createdName,
+              password: 'something-new'
+            }.to_json
+          )
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:user][:name]).to eq createdName
+          expect(resp.parsed_response_body[:user][:id]).to eq CREATED_ID
+        end
+      end
+    end
+
+    describe "when multiple users exists" do
+      TEST_USER_ID_2 = 'inspec-user-2'
+      TEST_USER_2 = {
+        id: TEST_USER_ID_2,
+        name: "display name 2",
+        password: "chefautomate"
+      }
+
+      before(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users",
+          http_method: 'POST',
+          request_body: TEST_USER.to_json
+        )
+        expect(resp.http_status).to eq 200
+
+        resp = automate_api_request("/apis/iam/v2beta/users",
+          http_method: 'POST',
+          request_body: TEST_USER_2.to_json
+        )
+        expect(resp.http_status).to eq 200
+      end
+
+      after(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users/#{CUSTOM_USER_ID}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+
+        resp = automate_api_request("/apis/iam/v2beta/users/#{TEST_USER_ID_2}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+      end
+
+      describe "GET /iam/v2beta/users/" do
+        it "returns the list of users" do
+          resp = automate_api_request("/apis/iam/v2beta/users")
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:users].length).to eq 3
+          expect(resp.parsed_response_body[:users][0][:id]).to match(/#{TEST_USER_ID_2}|#{CUSTOM_USER_ID}|#{ADMIN_USER_ID}/)
+          expect(resp.parsed_response_body[:users][1][:id]).to match(/#{TEST_USER_ID_2}|#{CUSTOM_USER_ID}|#{ADMIN_USER_ID}/)
+          expect(resp.parsed_response_body[:users][2][:id]).to match(/#{TEST_USER_ID_2}|#{CUSTOM_USER_ID}|#{ADMIN_USER_ID}/)
+        end
+      end
+    end
+
+    describe "when a user exists" do
+      before(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users",
+         http_method: 'POST',
+         request_body: TEST_USER.to_json
+       )
+       expect(resp.http_status).to eq 200
+     end
```

Comment:
I should think it is more realistic to start with multiple users (i.e. your list of 3 users) rather than just a single user in the DB. That makes the test stronger because rather than getting the right data because it is the _only_ one, instead it is because it is the _correct_ one.